### PR TITLE
[sidekiq] adding integration

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

Add the sidekiq integration.

Originally written by @bmarini (https://github.com/DataDog/dd-agent/pull/2066)

### Motivation

This adds a metric for the length of each sidekiq queue. This also supports collecting stats for queues when using reliable fetch, where each queue has a sub queues per worker.

The following stats are collected:

```
sidekiq.processed
sidekiq.failed
sidekiq.scheduled_size
sidekiq.retry_size
sidekiq.dead_size
sidekiq.processes_size
sidekiq.workers_size

Per queue:
  sidekiq.queue.length tag=queue:<queue name>
  sidekiq.queue.latency tag=queue:<queue name>
```

### CODE REVIEW

**Comment by irabinovitch** [comment]
@bmarini   Let us know if you need a hand with getting tests passing, etc. Feel free to stop by our office hours as well if we can help. Next one is this morning at 10:30 PT / 13:30 EST.

**Comment by Mange** [comment]
Any update on this? :-)

**Comment by jewilmeer** [comment]
I've tweaked the PR and got it working with these commits: [#1](https://github.com/DataDog/integrations-extras/pull/62/commits/ee731d6a6dd766d3236f58324af617e67e968d38) [#2](https://github.com/DataDog/integrations-extras/pull/62/commits/7502effc2cb971eb0411bc780a4471eb78266027)

**Comment by nickfishman** [comment]
I'm also curious about the status of this. My team is considering whether to start using DataDog, and this would be a really nice plus.

It looks like this might be really close after work by @jewilmeer in #62. Is this waiting on anything else?

**Comment by jeremy-lq** [comment]
@bmarini has indicated that this code has been working in production for over a year but that he doesn't currently have the time to be the maintainer. Is anyone else interested in becoming the maintainer for this integration?

**Comment by Mange** [comment]
Are you looking for maintainers in the community, or internally at DD?

**Comment by jeremy-lq** [comment]
@Mange integrations-extras are community maintained. Would you be interested in being the maintainer for the sidekiq integration? As mentioned, the initial submitter has indicated that this code has been working in production for over a year but that he doesn't currently have the time to be the maintainer. If you have any questions, let me know.

File: sidekiq/check.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        local scheduled_size = redis.call("ZCARD", KEYS[3])
+        local retry_size = redis.call("ZCARD", KEYS[4])
+        local dead_size = redis.call("ZCARD", KEYS[5])
+        local processes_size = redis.call("SCARD", KEYS[6])
+        local processes = redis.call("SMEMBERS", KEYS[6])
+
+        local workers_size = 0
+        for i, pkey in ipairs(processes) do
+            local nspkey = ARGV[1] == "None" and pkey or (ARGV[1] .. ":" .. pkey)
+            workers_size = workers_size + tonumber(redis.call("HGET", nspkey, "busy"))
```
Comment (bronislav) [inline sidekiq/check.py]: There is an minor issue with this line.

When there is no busy workers `redis.call("HGET", nspkey, "busy")` returns `(nil)`. And this causes exception:
```
@user_script:13: user_script:13: attempt to perform arithmetic on a nil value
```

Comment (bronislav) [reply sidekiq/check.py]: Any suggestions how it could be handled?

**Comment by collimarco** [comment]
Any updates on this? We are considering to move to DataDog and this feature would be really important for us.

**Comment by irabinovitch** [comment]
@collimarco This integration works but needs some love around testing. If you'd like to give it a go, we'd love to hear your feedback.

**Comment by bronislav** [comment]
@irabinovitch Please pay attention to my review comment. When there is no busy workers, check will be failed.

As a workaround, I've changed it to:
```lua
(tonumber(redis.call("HGET", nspkey, "busy")) or 0)
```

**Comment by jeremy-lq** [comment]
@bronislav Thanks for the review. This integration doesn't currently have a community maintainer, but if you or @collimarco are interested, let me know.

**Comment by bronislav** [comment]
@jeremy-lq I'm using it in prod right now. And I could maintain it time to time, I think. What is the requirements?

**Comment by jeremy-lq** [comment]
Thanks, @bronislav. To get the integration merged, it currently needs some
test improvements. From there it's mostly a matter of keeping an eye on bug
reports and feature requests from time to time. If you have any additional
questions, let me know.

**Comment by collimarco** [comment]
+1

**Comment by ryanmaclean** [comment]
Sorry for bumping this, and I can't offer stewardship, but I would certainly benefit from it!

**Review by victorcoder — COMMENTED** [review]
Hi @jeremy-lq, bumping this again, these metrics are very useful, we can take care of the integration maintenance, we use it extensively in out platform, WTYT?

**Comment by jeremy-lq** [comment]
@victorcoder There is a sidekiq integration in integrations-core now: https://github.com/DataDog/integrations-core/tree/master/sidekiq - if there are metrics missing from it, let us know.

**Comment by victorcoder** [comment]
@jeremy-lq that doesn't work for us, the integration needs Sidekiq Pro (>= 3.6) or Enterprise (>= 1.1.0) and we still have some legacy Rails services that we can not upgrade. Aside from that, the metric we are interested in is `sidekiq.enqueued` only available in Sidekiq Enterprise, and we are Sidekiq Pro customers only.

I see the value of this community integration for those use cases.

**Pull request closed without merging**